src/dfu/dfu_main.cpp

```diff
@@ -0,0 +1,1002 @@
+            {
+                add_command(dedicated_commands, command_name);
+            }
+        }
+    }
+    if (dfu_controller_servicer_resid == -1)
+    {
+        cerr << "Error: DFU_CONTROLLER_SERVICER_RESID not set. Check the YAML file: " << yaml_file_full_path << endl;
+        exit(HOST_APP_ERROR);
+    }
+}
+
+void check_verbose_mode(int * argc, char ** argv){
+    opt_t * use_opt = option_lookup("--verbose", options, num_options);
+    size_t index = argv_option_lookup(*argc, argv, use_opt);
+    if(index != 0)
+    {
+        cout << "Verbose mode enabled" << endl;
+        verbose_mode = 1;
+        remove_opt(argc, argv, index, 1);
+    }
+}
+
+uint16_t check_upload_start(int * argc, char ** argv){
+    opt_t * use_opt = option_lookup("--upload-start", options, num_options);
+    size_t index = argv_option_lookup(*argc, argv, use_opt);
+    // return -1 if the upload-start option is not found
+    if (index == 0) {
+        return INVALID_TRANSPORT_BLOCK_NUM;
+    }
+    if (index + 1 >= *argc)
+    {
+        cerr << "Error: missing block number" << endl;
+        exit(HOST_APP_ERROR);
+    }
+    string block_number_str = argv[index + 1];
+    if (!isdigit(block_number_str[0])) {
+        cerr << "Error: Value for transport block is not an integer. Given value: " << block_number_str << endl;
+        exit(HOST_APP_ERROR);
+    }
+    if (stoi(block_number_str) >= INVALID_TRANSPORT_BLOCK_NUM)
+    {
+        cerr << "Error: Max value for transport block is " << INVALID_TRANSPORT_BLOCK_NUM-1 << ". Given value: " << block_number_str << endl;
+        exit(HOST_APP_ERROR);
+    }
+    remove_opt(argc, argv, index, 2);
+
+    return stoi(block_number_str);
+}
+int main(int argc, char ** argv)
+{
+    if(argc == 1)
+    {
+        cout << "Use --help to get the list of options for this application." << endl
+        << "Or use --list-commands to print the list of commands and their info." << endl;
+        return 0;
+    }
+    YAML::Node config;
+    int* i2c_info = new int[1];
+    int* spi_info = new int[2];
+    string yaml_file_name;
+
+    // Check optional arguments
+    check_verbose_mode(&argc, argv);
+    uint16_t start_block_number = check_upload_start(&argc, argv);
+
+    // Load YAML file with transport settings
+    yaml_file_name = get_executable_path() + "/transport_config.yaml";
+    if (verbose_mode) {
+        cout << "Parsing YAML file " << yaml_file_name << endl;
+    }
+    if (is_file_found(yaml_file_name)) {
+        config = YAML::LoadFile(yaml_file_name);
+        // Read I2C parameters from YAML file
+        i2c_info[0] = config["I2C_ADDRESS"].as<int>();
+        spi_info[0] = config["SPI_MODE"].as<int>();
+        spi_info[1] = 1024;
+    } else {
+        cerr << "Error: File \'" << yaml_file_name << "\' not found" << endl;
+        exit(HOST_APP_ERROR);
+    }
+
+    // Load dynamic library with transport drivers
+    string device_dl_name = get_device_lib_name(&argc, argv, options, num_options);
```

Comment:
If the user input protocol is not valid, this API will return a string "Will use USB by default" and use I2C

Reply:
I updated the printout. Now it says for example:

`Will use default driver: device_usb`